src/psyclone/domain/lfric/lfric_builtins.py

```diff
@@ -164,9 +175,47 @@ def __init__(self):
         self._idx_name = None
         super().__init__()
 
+    @staticmethod
     @abc.abstractmethod
-    def __str__(self):
-        ''' Must be overridden by sub class. '''
+    def metadata():
+        ''' Must be overridden by subclass. '''
+
+    @classmethod
+    def _builtin_metadata(cls, meta_args):
+        '''Utility to take meta_args metadata and return LFRic kernel metadata
+        for a builtin. Assumes the metadata describes a builtin kernel
+        that operates on a dof and that the naming protocol uses the
+        name of the metadata type and adds _code to it for the name of
+        the subroutine.
+
+        :param meta_args: a list of meta args metadata.
+        :type meta_args: List[subclass of \
+            :py:class:`psyclone.domain.lifric.kernel.CommonArgMetadata`]
+
+        :returns: lfric kernel metadata for this builtin.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.LFRicKernelMetadata`
+
+        '''
+        return LFRicKernelMetadata(
+            meta_args=meta_args,
+            operates_on="dof",
+            procedure_name=f"{cls._case_name}_code",
+            name=cls._case_name)
+
+    def __str__(self):
+        if not self._datatype:
+            raise NotImplementedError(
+                "An LFRicBuiltin should be overridden by a subclass that "
+                "sets the value of _datatype, but _datatype is not set.")
+        metadata = self.metadata()
+        plural = ""
+        # Builtins are currenty limited to fields and scalars but add
+        # in a check for field-vectors as well for future proofing.
+        if len(metadata.meta_args_get([
+                FieldArgMetadata, FieldVectorArgMetadata])) > 1:
+            plural = "s"
+        return (f"Built-in: {self._case_name} ("
+                f"{self._datatype}-valued field{plural})")
```

Comment:
Not in this line, but `pylint` returns a warning: `lfric_builtins.py:247:4: W0221: Number of parameters was 4 in 'BuiltIn.load' and is now 3 in overridden 'LFRicBuiltIn.load' method (arguments-differ)`

Reply:
This is a pre-existing issue to do with a difference between the `builtin.load()` and `lfric.load()` interfaces, where the former also passes in the type of the arguments class so that the base class can work independently of API. This should all change eventually with the new PSyIR metadata approach.